torch/_subclasses/meta_utils.py

```diff
@@ -429,20 +740,29 @@ def symint_visitor_fn(s):
                 )
                 attrs, _ = fake_t.__tensor_flatten__()
                 for attr in attrs:
-                    real_to_fake_mapping[getattr(t, attr)] = getattr(fake_t, attr)
+                    real_to_fake_mapping[t.attrs[attr].id] = getattr(fake_t, attr)
 
             def tensor_visitor_fn(
-                visited_t, shape_env=shape_env, callback=callback, source=source
+                visited_t: torch.Tensor,
+                shape_env=shape_env,
+                callback=callback,
+                source=source,
             ):
                 # It's possible to close over an undefined tensor (e.g. NJT's lengths).
                 if visited_t is None:
                     return None
 
+                # NB: visited_t being a Tensor here is very naughty!  Should
+                # have already been described
+
```

Comment:
Why is this is not naughty ? How difficult would it be to fix in this PR?

Reply:
It is naughty. The view function is closing over real tensors, so we're failing to enforce full separation here. This is the view func gap; it's fixed by reifying the view func and then describing all the closed over values in the explicitly represented closure.